src/backend/utils/adt/ruleutils.c

```diff
@@ -3175,6 +3177,11 @@ print_function_arguments(StringInfo buf, HeapTuple proctup,
 	{
 		Datum		proargdefaults;
 		bool		isnull;
+		char		*langname = get_language_name(proc->prolang, true);
+
```

Comment:
logically, if we are duplicating most of the lost, it might be better if we inject our logic from the beginning (e.g line 3169)  and only check dialect.

Reply:
Dialect check won't work here as this particular function gets called to dump a function in PG dialect, so that's why I am only checking for function's language but I moved this hook to the beginning.